src/librustc_typeck/diagnostics.rs

```diff
@@ -170,6 +170,38 @@ Since `return;` is just like `return ();`, there is a mismatch between the
 function's return type and the value being returned.
 "##,
 
+E0070: r##"
+The left-hand side of an assignment operator must be an lvalue expression. An
+lvalue expression represents a memory location and includes item paths (ie,
+namespaced variables), dereferences, indexing expressions, and field references.
+
+More details can be found here:
+https://doc.rust-lang.org/reference.html#lvalues,-rvalues-and-temporaries
+
+Now, we can go further. Here are some bad examples:
+```
+const SOME_CONST : i32 = 12;
+
+fn some_function() {
+    SOME_CONST = 14; // error !
+    1=3; // error 
+}
+```
+
+And now let's give good examples:
+
+```
+static mut SOME_NOT_CONST : i32 = 12;
+
+fn some_function() {
+    SOME_NOT_CONST = 14; // that's good !
+    let mut i : i32 = 1;
+    
+    i = 3; // that's good !
```

Comment:
I just wanted to give multiple example cases, I think it's always better. People could think that only const variables could be concerned by this error, I want to avoid that.